### DISCUSSION TITLE
Problem with migrate the permission_table

### DISCUSSION THREAD

**Discussion by ghost** [spatie/laravel-permission#2192] [General]
Hello,

I tried to generate the database on the server via php artisan migrate. 
Everything worked except for the permission_table. 
I get the following error message here (see attachment).
Can you maybe help me?

![Error](https://user-images.githubusercontent.com/85445509/193209191-688eb72f-07c8-4b39-af4a-0444a5a37d84.PNG)

**Comment by erikn69** [comment]
> Can you maybe help me?

Without laravel version, package version, php version, db version, I don't think anyone can help you

**Comment by ghost** [comment]
Sorry, my mistake.

-  Laravel Version => 9.32
- PHP Version => 8.1
- yajra/laravel-oci8´=> 9.0.3
- spatie/laravel-permission => 5.5.5

I use a Oracle-Database

**Comment by erikn69** [comment]
[error-ora-00972](https://kb.tableau.com/articles/issue/error-ora-00972)
[ora-00972](https://www.google.com/search?q=ora-00972&oq=ora-00972&aqs=chrome..69i57.4100j0j1&sourceid=chrome&ie=UTF-8)

Change names for primary and foreign keys
```php
Schema::create($tableNames['model_has_permissions'], function (Blueprint $table) use ($tableNames, $columnNames, $teams) {
    $table->unsignedBigInteger(PermissionRegistrar::$pivotPermission);

    $table->string('model_type');
    $table->unsignedBigInteger($columnNames['model_morph_key']);
    $table->index([$columnNames['model_morph_key'], 'model_type'], 'mhp_mid_mtype_index');

    $table->foreign(PermissionRegistrar::$pivotPermission, 'mhp_permission')
        ->references('id') // permission id
        ->on($tableNames['permissions'])
        ->onDelete('cascade');
    if ($teams) {
        $table->unsignedBigInteger($columnNames['team_foreign_key']);
        $table->index($columnNames['team_foreign_key'], 'mhp_tid_index');

        $table->primary([$columnNames['team_foreign_key'], PermissionRegistrar::$pivotPermission, $columnNames['model_morph_key'], 'model_type'],
            'mhp_permission_mtype_key');
    } else {
        $table->primary([PermissionRegistrar::$pivotPermission, $columnNames['model_morph_key'], 'model_type'],
            'mhp_permiss_mtype_key');
    }

});
```
